src/v/security/gssapi_authenticator.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+#include "security/acl.h"
+#include "security/credential_store.h"
+#include "security/gssapi.h"
+#include "security/logger.h"
+#include "security/sasl_authentication.h"
+#include "security/scram_algorithm.h"
+#include "security/types.h"
+
+namespace security {
+
+void display_status(const char* msg, OM_uint32 maj_stat, OM_uint32 min_stat);
+std::string display_ctx_flags(OM_uint32 flags);
+
+class gssapi_authenticator final : public sasl_mechanism {
+    enum class state { init = 0, more, ssfcap, ssfreq, complete, failed };
+    friend std::ostream&
+    operator<<(std::ostream& os, gssapi_authenticator::state const s);
+
+public:
+    static constexpr const char* name = "GSSAPI";
+
+    explicit gssapi_authenticator(credential_store& credentials)
+      : _credentials{credentials} {}
+
+    result<bytes> authenticate(bytes_view) override;
+
+    bool complete() const override {
+        vlog(seclog.info, "gss complete: {}", _state);
+        return _state == state::complete;
+    }
+    bool failed() const override {
+        vlog(seclog.info, "gss failed: {}", _state);
+        return _state == state::failed;
+    }
+
+    const security::acl_principal& principal() const override {
+        vlog(seclog.info, "gss principal: {} {}", _principal, _state);
```

Comment:
curious to have the vlog in what seems like a simple accessor?

Reply:
Removed